pkg/storage/factory_test.go

```diff
@@ -91,56 +137,102 @@ func TestNamedStores(t *testing.T) {
 	t.Run("period config referring to configured named store", func(t *testing.T) {
 		store, err := NewStore(cfg, config.ChunkStoreConfig{}, schemaConfig, limits, cm, nil, util_log.Logger)
 		require.NoError(t, err)
+		defer store.Stop()
 
-		store.Stop()
+		// FSObjectClient creates the configured dir on init, ensure that correct cfg is picked by checking for it.
+		_, err = os.Stat(cfg.NamedStores.Filesystem["named-store"].Directory)
```

Comment:
You should first check that this directory does not exist before instantiating the store, to account for false-positives.